pkg/kubelet/kubelet.go

```diff
@@ -3282,7 +3311,21 @@ func GetPhase(spec *api.PodSpec, info []api.ContainerStatus) api.PodPhase {
 
 func (kl *Kubelet) generateAPIPodStatus(pod *api.Pod, podStatus *kubecontainer.PodStatus) api.PodStatus {
 	glog.V(3).Infof("Generating status for %q", format.Pod(pod))
+
+	// any pod that requires eviction must be added to the list to sync.
+	attrs := &lifecycle.PodEvictAttributes{Pod: pod}
+	for _, podEvictor := range kl.podEvictors {
+		if result := podEvictor.Evict(attrs); result.Evict {
```

Comment:
@pmorie ^^^ see above comment, this is reflective of our discussion last friday.

We have 4 different channels that can ultimately result in a pod being synced, and experience has proven to me that doing anything outside of those 4 channels is asking for bugs.

- updates -> used to drive pod workers to realize changes driven by the apiserver or kubelet local file source
- pleg -> used to respond to events from container runtime
- housekeeping -> used to do any clean-ups for orphaned stuff in kubelet
- sync -> checks workqueue that was filled by pod_workers to tell kubelet when it should be resynced, and handle active deadline killing every 1s

For purposes of eviction, I think we need to decide which channel is the right channel to drive eviction.  I think the sync channel is proper, but agree we shouldn't have to wait for the housekeeping interval to ultimately kill the pod (per the comment above).  Would you agree that trying to have internal kubelet modules speak directly to the pod_worker to handle work not driven from the updates channel is confusing?  At the end of the day, no matter the path, we call kubelet.syncPod()